src/index.tsx

```diff
@@ -74,3 +76,19 @@ function getApiKey() {
     // use graphhopper api key from url or try using one from the config
     return url.searchParams.has('key') ? url.searchParams.get('key') : config.keys.graphhopper
 }
+
+function parseURLBBox() : Bbox | null {
```

Comment:
Ultimately this could be in the ```NavBar``` component as well. Once you update the viewport state by dispatching an action, you could put this logic there.